beacon-chain/db/kv/blocks.go

```diff
@@ -132,19 +132,61 @@ func (k *Store) Blocks(ctx context.Context, f *filters.QueryFilter) ([]*ethpb.Be
 func (k *Store) BlockRoots(ctx context.Context, f *filters.QueryFilter) ([][]byte, error) {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.BlockRoots")
 	defer span.End()
-	blocks, err := k.Blocks(ctx, f)
-	if err != nil {
-		return nil, err
-	}
-	roots := make([][]byte, len(blocks))
-	for i, b := range blocks {
-		root, err := ssz.SigningRoot(b)
+	blockRoots := make([][]byte, 0)
+	err := k.db.View(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(blocksBucket)
+
+		// If no filter criteria are specified, return all block roots.
```

Comment:
what is the use case for this? It seems expensive to allow a db call that returns the whole db.

Reply:
Will return error